tests/credential_plugins_test.py

```diff
@@ -162,3 +165,35 @@ def test_tss_import(self) -> None:
         ):
             # assert this module as opposed to older thycotic.secrets.server
             assert cls.__module__ == 'delinea.secrets.server'
+
+
+def test_aim_sensitive_traceback_masked(mocker: MockerFixture) -> None:
+    """Ensure that the sensitive information is not leaked in the traceback."""
+    my_response = requests.Response()
+    my_response.status_code = 404
+    my_response.url = 'not_found'
+
+    aim.requests.get = mocker.Mock(name='aim_request')
```

Comment:
I didn't notice this before, but this is mutating the process-global state and doesn't restore it after. To fix this, either use `monkeypatch.setattr()` or `mocker.patch` both of which are able to undo the changes..

Reply:
Marking this as resolved.